### PR TITLE
Remove 'torch' from deps and warn instead

### PR DESCRIPTION
**Pull request by marbre**
Instead of enforcing the installation for 'torch' as a dependency, error if 'torch' cannot be imported and point the user to how to install.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -107,7 +107,6 @@ def initialize_options(self):
         f"numpy{get_version_spec('numpy')}",
         f"iree-base-compiler{get_version_spec('iree-base-compiler')}",
         f"iree-base-runtime{get_version_spec('iree-base-runtime')}",
-        "torch>=2.3.0",
```
Comment (ScottTodd) [inline setup.py:110]: Let's update the docs too.

https://github.com/iree-org/iree-turbine?tab=readme-ov-file#quick-start-for-users
https://github.com/iree-org/iree-turbine?tab=readme-ov-file#install-pytorch-for-your-system

Specifically mention that installing `torch` is the user's responsibility and drop these comments:

> The above does install some CUDA/cuDNN packages which are unnecessary for most usage.

> If no explicit action is taken, the default PyTorch version will be installed.

Comment (ScottTodd) [reply setup.py:110]: In the IREE docs we can add a Linux/macOS/Windows branch in the docs here too: https://iree.dev/guides/ml-frameworks/pytorch/#prerequisites. I'll send a PR for that.

Comment (marbre) [reply setup.py:110]: > In the IREE docs we can add a Linux/macOS/Windows branch in the docs here too: https://iree.dev/guides/ml-frameworks/pytorch/#prerequisites. I'll send a PR for that.

Thanks! Updated the docs here, excuse me initialy missing this.

File: README.md [hunk 2]
```diff
@@ -32,6 +32,14 @@ Discord server).
 
 1. Install from PyPI:
 
+    Install a torch version that fulfills your needs:
+
+    ```bash
+    # Fast installation of torch with just CPU support.
+    # See other options at https://pytorch.org/get-started/locally/
+    pip install -r pytorch-cpu-requirements.txt
+    ```
```
Comment (ScottTodd) [inline README.md:41]: These user instructions don't assume that you have the repository checked out yet, so I don't think we should depend on this file (and it only works on Linux right now)

File: setup.py (hunk 1 above)
Comment (ScottTodd) [reply setup.py:110]: Sent https://github.com/iree-org/iree/pull/19493 for https://iree.dev/guides/ml-frameworks/pytorch/#prerequisites

File: README.md (hunk 2 above)
Comment (marbre) [reply README.md:41]: Good catch. Updated again.

Comment (ScottTodd) [reply README.md:41]: Oh, you updated the section below but not this first user section? The text down there LGTM, but this user section right at the top of the docs shouldn't assume you have the git repo cloned.

**Review by ScottTodd — APPROVED** [review]
One more nit

**Pull request merged**